lib/src/winrt/internal/vector_helper.dart

```diff
@@ -0,0 +1,151 @@
+    } else if (isSubtypeOfWinRTEnum<T>()) {
+      list = _toList_enum();
+    } else {
+      list = _toList_COMObject();
+    }
+
+    return unmodifiable ? List.unmodifiable(list) : list;
+  }
+
+  List<String> _toList_String() {
+    final pArray = calloc<HSTRING>(length);
+
+    try {
+      getManyCallback(0, length, pArray);
+      return pArray.toList(length: length);
+    } finally {
+      free(pArray);
+    }
+  }
+
+  List<T> _toList_COMObject() {
+    final pArray = allocator<COMObject>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(creator!, length: length);
+  }
+
+  List<T> _toList_enum() {
+    switch (intType) {
+      case Uint32:
+        return _toList_enum_Uint32();
+      default:
+        return _toList_enum_Int32();
+    }
+  }
+
+  List<T> _toList_enum_Int32() {
+    final pArray = allocator<Int32>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length).map((e) => enumCreator!(e)).toList();
+  }
+
+  List<T> _toList_enum_Uint32() {
+    final pArray = allocator<Uint32>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length).map((e) => enumCreator!(e)).toList();
+  }
+
+  List<int> _toList_int() {
+    switch (intType) {
+      case Int16:
+        return _toList_int_Int16();
+      case Int64:
+        return _toList_int_Int64();
+      case Uint8:
+        return _toList_int_Uint8();
+      case Uint16:
+        return _toList_int_Uint16();
+      case Uint32:
+        return _toList_int_Uint32();
+      case Uint64:
+        return _toList_int_Uint64();
+      default:
+        return _toList_int_Int32();
+    }
+  }
+
+  List<int> _toList_int_Int16() {
+    final pArray = allocator<Int16>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Int32() {
+    final pArray = allocator<Int32>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Int64() {
+    final pArray = allocator<Int64>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Uint8() {
+    final pArray = allocator<Uint8>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Uint16() {
+    final pArray = allocator<Uint16>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Uint32() {
+    final pArray = allocator<Uint32>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+
+  List<int> _toList_int_Uint64() {
+    final pArray = allocator<Uint64>(length);
+    getManyCallback(0, length, pArray);
+    return pArray.toList(length: length);
+  }
+}
+
+const supportedIntTypes = [Int16, Int32, Int64, Uint8, Uint16, Uint32, Uint64];
```

Comment:
I couldn't help but notice that `Int8` is missing. is that not a viable option? If not, then you might consider a comment with a link to relevant WinRT documentation to avoid confusion.

Reply:
`Int8` is not supported on the WinRT Type system: https://docs.microsoft.com/en-us/uwp/winrt-cref/winrt-type-system#fundamental-types

I added a comment with the link above explaining this.

Reply:
Great - thanks.